object-cache.php

```diff
@@ -508,64 +554,171 @@ function switch_to_blog( $blog_id ) {
+			if ( ! element ) {
+				return;
+			}
+
+			// Hide all element with `idPrefix` if given. Used to display only one element at a time.
+			if ( hidePrefix ) {
+				var groupStats = document.querySelectorAll( '[id^=\"' + hidePrefix + '\"]' );
+				Array.prototype.forEach.call(
+					groupStats,
+					function ( element ) {
+					    element.style.display = 'none';
+					}
+				);
+			}
+
+			// Toggle the one we clicked.
+			if ( 'none' === element.style.display ) {
+				element.style.display = 'block';
+			} else {
+				element.style.display = 'none';
+			}
 		}
+		</script>
+		";
+	}
+
+	function stats() {
+		$this->js_toggle();
 
-		echo "<p>\n";
+		echo '<h2><span>Total memcache query time:</span>' . number_format( sprintf( '%0.1f', $this->time_total * 1000 ), 1, '.', ',' ) . 'ms</h2>';
+		echo "\n";
+		echo '<h2><span>Total memcache size:</span>' . $this->human_filesize( $this->size_total ) . '</h2>';
+		echo "\n";
 
 		foreach ( $this->stats as $stat => $n ) {
-			echo '<strong>' . esc_html( $stat ) . '</strong>' . esc_html( $n );
-			echo "<br/>\n";
+			if ( empty( $n ) ) {
+				continue;
+			}
+
+			echo '<h2>';
+			echo $this->colorize_debug_line( "$stat $n" );
+			echo '</h2>';
 		}
 
-		echo "</p>\n";
-		echo '<h3>Memcached:</h3>';
+		echo "<ul class='debug-menu-links'>\n";
+		$groups = array_keys( $this->group_ops );
+		usort( $groups, 'strnatcasecmp' );
 
-		foreach ( $this->group_ops as $group => $ops ) {
-			if ( ! isset( $_GET['debug_queries'] ) && 500 < count( $ops ) ) {
-				$ops = array_slice( $ops, 0, 500 );
-				echo "<big>Too many to show! <a href='" . esc_url( add_query_arg( 'debug_queries', 'true' ) ) . "'>Show them anyway</a>.</big>\n";
-			}
+		$active_group = $groups[0];
+		// Always show `slow-ops` first
+		if ( in_array( 'slow-ops', $groups ) ) {
+			$slow_ops_key = array_search( 'slow-ops', $groups );
+			$slow_ops = $groups[ $slow_ops_key ];
+			unset( $groups[ $slow_ops_key ] );
+			array_unshift( $groups, $slow_ops );
+			$active_group = 'slow-ops';
+		}
 
-			echo '<h4>' . esc_html( $group ) . ' commands</h4>';
+		$total_ops = 0;
+		$group_titles = array();
+		foreach ( $groups as $group ) {
+			$group_name = $group;
+			if ( empty( $group_name ) ) {
+				$group_name = 'default';
+			}
+			$group_ops = count( $this->group_ops[ $group ] );
+			$group_size = $this->human_filesize( array_sum( array_map( function ( $op ) { return $op[2]; }, $this->group_ops[ $group ] ) ) );
+			$group_time = number_format( sprintf( '%0.1f', array_sum( array_map( function ( $op ) { return $op[3]; }, $this->group_ops[ $group ] ) ) * 1000 ), 1, '.', ',' );
+			$total_ops += $group_ops;
+			$group_title = "{$group_name} [$group_ops][$group_size][{$group_time}ms]";
+			$group_titles[ $group ] = $group_title;
+			echo "\t<li><a href='#' onclick='memcachedToggleVisibility( \"object-cache-stats-menu-target-" . esc_attr( $group ) . "\", \"object-cache-stats-menu-target-\" );'>" . esc_html( $group_title ) . "</a></li>\n";
```

Comment:
Minor: would `esc_js` be a better choice here since this is inline javascript?

```suggestion
			echo "\t<li><a href='#' onclick='memcachedToggleVisibility( \"object-cache-stats-menu-target-" . ec_js( $group ) . "\", \"object-cache-stats-menu-target-\" );'>" . esc_html( $group_title ) . "</a></li>\n";
```

Reply:
Typo in `ec_js` 🙈